ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/assertions/CommandDeduplicationAssertions.scala

```diff
@@ -51,37 +55,93 @@ object CommandDeduplicationAssertions {
+          .previous(reportedHexOffset)
+          .map(offset => LedgerOffset.of(LedgerOffset.Value.Absolute(offset)))
+          .getOrElse(ledger.referenceOffset)
+        val reportedOffsetCompletionStreamRequest =
+          ledger.completionStreamRequest(offsetPreviousToReportedOffset)(submittingParty)
         if (completionResponse.completion.getStatus.code == Code.ALREADY_EXISTS.value) {
-          // Search for the first accepting completion with the same command ID since the beginning of the test case.
-          // Multiple consecutive accepting completions are not supported.
-          val completionStreamRequest = ledger.completionStreamRequest()(submittingParty)
-          WithTimeout(5.seconds)(
-            ledger
-              .findCompletion(completionStreamRequest)(c =>
-                c.commandId == c.commandId && c.getStatus.code == Code.OK.value
-              )
-          ).map { optAcceptedCompletionResponse =>
-            val acceptedCompletionResponse = assertDefined(
-              optAcceptedCompletionResponse,
-              s"No accepted completion with the command ID '${completionResponse.completion.commandId}' has been found",
-            )
-            assert(
-              acceptedCompletionResponse.offset.getAbsolute >= reportedOffset,
-              s"No accepted completion with the command ID '${completionResponse.completion.commandId}' after the reported offset $reportedOffset has been found",
-            )
-            assert(
-              acceptedCompletionResponse.offset.getAbsolute < completionResponse.offset.getAbsolute,
-              s"No accepted completion with the command ID '${completionResponse.completion.commandId}' before the completion's offset ${completionResponse.offset} has been found",
-            )
-          }
+          assertReportedOffsetForDuplicateSubmission(
+            reportedHexOffset,
+            completionResponse,
+            ledger,
+            reportedOffsetCompletionStreamRequest,
+          )
         } else {
-          fail("This case is not yet supported by this assertion")
+          assertReportedOffsetForAcceptedSubmission(
+            reportedHexOffset,
+            requestedDuration,
+            completionResponse,
+            ledger,
+            reportedOffsetCompletionStreamRequest,
+          )
         }
       case DurationSupport.Unrecognized(_) =>
         fail("Unrecognized deduplication duration support")
     }
   }
 
+  private def assertReportedOffsetForDuplicateSubmission(
+      reportedOffset: Ref.HexString,
+      completionResponse: CompletionResponse,
+      ledger: ParticipantTestContext,
+      reportedOffsetCompletionStreamRequest: CompletionStreamRequest,
+  )(implicit executionContext: ExecutionContext) = {
+    WithTimeout(5.seconds)(
+      ledger
+        .findCompletion(reportedOffsetCompletionStreamRequest)(c =>
+          c.commandId == completionResponse.completion.commandId && c.getStatus.code == Code.OK.value
```

Comment:
Technically, command deduplication looks at the change ID (= application ID + actAs + commandId). It's probably OK though to just look at the command ID here.